app/Jobs/FetchAd.php

```diff
@@ -67,50 +74,45 @@ private function fetch(stdClass $source): array
 
         do {
             $counter = 0;
-            [$fetchDone , $items , $currentPage , $nextPageUrl , $lastPage, $resultText] = $this->adFetcher->fetchAd($fetchUrl);
-            if(!$fetchDone){
+            [$fetchDone, $items, $currentPage, $nextPageUrl, $lastPage, $resultText] = $this->adFetcher->fetchAd($fetchUrl);
+            if (!$fetchDone) {
                 $failedPages++;
                 continue;
             }
 
             if (empty($items)) {
                 continue;
```

Comment:
It may cause infinite loop, put some think behind it please.

Reply:
done